core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -3686,4 +3742,43 @@ Sql assertRowsUnique(boolean[] ignoreNulls, Matcher<Boolean> matcher,
     }
 
   }
+
+  /**
+   * Test class for janino metadata path.
+   */
+  @Nested
```

Comment:
I am not sure if this is an appropriate use of the `@Nested` annotation. Citing the [JUnit5 user guide](https://junit.org/junit5/docs/current/user-guide/#writing-tests-nested):
`Only non-static nested classes (i.e. inner classes) can serve as @Nested test classes`.

I think the test run in this case cause you are using also inheritance but again I am not sure if those were meant to be combined. Instead of relying on test inheritance or `@Nested`, [parameterized tests](https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests) would better fit here I think.

Reply:
Yeah, I was torn on making it parameterized given the sheer amount of minor changes. 

I just went through and did so. It quadrupled the size of the patch but I agree it is more idiomatic.

Reply:
Oh, sorry about that. It wasn't a "must do" comment from my side but thanks for taking care of it.